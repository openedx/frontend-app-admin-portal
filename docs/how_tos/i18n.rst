####################
React App i18n HOWTO
####################

This document has moved to the frontend-platform repo: https://github.com/openedx/frontend-platform/blob/master/docs/how_tos/i18n.rst
