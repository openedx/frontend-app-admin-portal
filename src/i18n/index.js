import { messages as paragonMessages } from '@edx/paragon';

export default [
  paragonMessages,
];
