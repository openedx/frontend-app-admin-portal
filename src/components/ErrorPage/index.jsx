import React from 'react';
import PropTypes from 'prop-types';
import { Helmet } from 'react-helmet';

import { Alert } from '@edx/paragon';
import { Cancel as ErrorIcon } from '@edx/paragon/icons';
import { FormattedMessage } from '@edx/frontend-platform/i18n';

import NotFoundPage from '../NotFoundPage';
import ForbiddenPage from '../ForbiddenPage';

function renderErrorComponent(status, message) {
  const errorMessage = message || 'An unknown error has occured.';
  if (status === 404) {
    return <NotFoundPage />;
  }

  if (status === 403) {
    return <ForbiddenPage />;
  }

  return (
    <>
      <Helmet>
        <title>Error</title>
      </Helmet>
      <div className="row mt-4">
        <div className="col">
          <Alert
            variant="danger"
            icon={ErrorIcon}
          >
            <Alert.Heading><FormattedMessage id="adminPortal.error.title" defaultMessage="Error" /></Alert.Heading>
            {errorMessage}
          </Alert>
        </div>
      </div>
    </>
  );
}

const ErrorPage = (props) => {
  const { status, message } = props;
  return (
    <main role="main">
      <div className="container-fluid">
        {renderErrorComponent(status, message)}
      </div>
    </main>
  );
};

ErrorPage.propTypes = {
  status: PropTypes.number,
  message: PropTypes.string,
};

ErrorPage.defaultProps = {
  status: null,
  message: '',
};

export default ErrorPage;
