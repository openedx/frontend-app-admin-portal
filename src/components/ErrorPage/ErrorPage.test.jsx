import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { IntlProvider } from '@edx/frontend-platform/i18n';

import ErrorPage from './index';

const ErrorPageWrapper = (props) => (
  <MemoryRouter>
    <IntlProvider locale="en">
      <ErrorPage {...props} />
    </IntlProvider>
  </MemoryRouter>
);

describe('<ErrorPage />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create((
        <ErrorPageWrapper status={500} message="Something went terribly wrong" />
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly for 404 errors', () => {
    const tree = renderer
      .create((
        <ErrorPageWrapper status={404} />
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly for 403 errors', () => {
    const tree = renderer
      .create((
        <ErrorPageWrapper status={403} />
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
