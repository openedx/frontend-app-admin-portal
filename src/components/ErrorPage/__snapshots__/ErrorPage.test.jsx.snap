// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorPage /> renders correctly 1`] = `
<main
  role="main"
>
  <div
    className="container-fluid"
  >
    <div
      className="row mt-4"
    >
      <div
        className="col"
      >
        <div
          className="fade alert-content alert alert-danger show"
          role="alert"
        >
          <span
            className="pgn__icon alert-icon"
          >
            <svg
              aria-hidden={true}
              fill="none"
              focusable={false}
              height={24}
              role="img"
              viewBox="0 0 24 24"
              width={24}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2Zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59Z"
                fill="currentColor"
              />
            </svg>
          </span>
          <div
            className="pgn__alert-message-wrapper"
          >
            <div
              className="alert-message-content"
            >
              <div
                className="alert-heading h4"
              >
                Error
              </div>
              Something went terribly wrong
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</main>
`;

exports[`<ErrorPage /> renders correctly for 403 errors 1`] = `
<main
  role="main"
>
  <div
    className="container-fluid"
  >
    <main
      role="main"
    >
      <div
        className="container-fluid mt-3"
      >
        <div
          className="text-center py-5"
        >
          <h1>
            403
          </h1>
          <p
            className="lead"
          >
            You do not have access to this page.
          </p>
          <p>
            For assistance, please contact the edX Customer Success team at
             
            <a
              className="pgn__hyperlink default-link standalone-link pgn__mailtolink"
              href="mailto:customersuccess@edx.org"
              onClick={null}
              target="_self"
            >
              customersuccess@edx.org
            </a>
            .
          </p>
        </div>
      </div>
    </main>
  </div>
</main>
`;

exports[`<ErrorPage /> renders correctly for 404 errors 1`] = `
<main
  role="main"
>
  <div
    className="container-fluid"
  >
    <main
      role="main"
    >
      <div
        className="container-fluid mt-3"
      >
        <div
          className="text-center py-5"
        >
          <h1>
            404
          </h1>
          <p
            className="lead"
          >
            Oops, sorry we can't find that page!
          </p>
          <p>
            Either something went wrong or the page doesn't exist anymore.
          </p>
        </div>
      </div>
    </main>
  </div>
</main>
`;
